src/Metrics.h

```diff
@@ -235,12 +237,27 @@ class TopN final : public Metric
     datasketches::frequent_items_sketch<T> _fi;
     size_t _top_count = 10;
     std::string _item_key;
+    double _percentile;
+
+    uint64_t _get_threshold() const
+    {
+        auto items = _fi.get_frequent_items(datasketches::frequent_items_error_type::NO_FALSE_NEGATIVES);
+        datasketches::kll_sketch<uint64_t> quantile;
+        for (uint64_t i = 0; i < std::min(_top_count, items.size()); i++) {
+            quantile.update(items[i].get_estimate());
+        }
+        if (quantile.is_empty()) {
```

Comment:
get_quantile throws it quantile is empty. that's why I needed to add this validation